app/models/assignment.rb

```diff
@@ -695,6 +655,18 @@ def get_detailed_csv_report
     end
   end
 
+  # Returns an array of arrays of the form [mark, max] for the corresponding criteria
+  def get_marks_list(submission)
+    get_criteria.first.nil? ? [] : get_criteria.first.get_marks_list(submission)
+  end
+
+  # Returns an array of the form ['', max] or ['', weight] depending on the criteria
+  # This is not a good name right now because we either return weight or max
+  # TODO This may change when unifying flexible and rubric
+  def get_max_mark_for_criteria
+    get_criteria.first.nil? ? ['', 0] : get_criteria.first.get_max_mark_for_criteria
```

Comment:
Calling `get_criteria.first.get_max_mark_for_criteria` makes no sense since it looks at all the criteria.

Generate the list here, not inside `get_criteria.first.get_max_mark_for_criteria``